cli/cook/subcommands/cat.py

```diff
@@ -21,6 +20,30 @@ def cat_for_instance(instance, sandbox_dir, _, path):
         sys.stderr.close()
         logging.exception(bpe)
 
+def kubectl_cat_instance_file(instance_uuid, _, path):
+    os.execlp('kubectl', 'kubectl',
+              'exec',
+              '-c', os.getenv('COOK_CONTAINER_NAME_FOR_JOB', 'required-cook-job-container'),
+              '-it', instance_uuid,
+              '--', 'cat', path)
+
+def cat_for_instance(instance, sandbox_dir_fn, cluster, path):
+    """
+    Outputs the contents of the Mesos sandbox path for the given instance.
+    When using Kubernetes, calls the exec command of the kubectl cli.
+    """
+    compute_cluster = instance["compute-cluster"]
+    compute_cluster_type = compute_cluster["type"]
+    compute_cluster_name = compute_cluster["name"]
+    if compute_cluster_type == "kubernetes":
+        kubectl_cat_instance_file_fn = plugins.get_fn('kubectl-cat-for-instance', kubectl_cat_instance_file)
+        cook_cluster_settings = http.get(cluster, 'settings', params={}).json()
+        compute_cluster_config = next(c for c in (s['config'] for s in cook_cluster_settings['compute-clusters']) if
+                                      c['compute-cluster-name'] == compute_cluster_name)
```

Comment:
Can we extract this into a function since we do it in a few places?

Reply:
i'll update